core/src/main/java/com/expedia/www/haystack/client/dispatchers/clients/GRPCAgentClient.java

```diff
@@ -0,0 +1,170 @@
+package com.expedia.www.haystack.client.dispatchers.clients;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.expedia.open.tracing.agent.api.DispatchResult;
+import com.expedia.open.tracing.agent.api.SpanAgentGrpc;
+import com.expedia.open.tracing.agent.api.SpanAgentGrpc.SpanAgentStub;
+import com.expedia.www.haystack.client.Span;
+import com.expedia.www.haystack.client.dispatchers.formats.Format;
+import com.expedia.www.haystack.client.dispatchers.formats.ProtoBufFormat;
+
+import io.grpc.ManagedChannel;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.stub.StreamObserver;
+
+public class GRPCAgentClient implements Client {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GRPCAgentClient.class);
+
+    private final Format<com.expedia.open.tracing.Span> format;
+    private final ManagedChannel channel;
+    private final SpanAgentStub stub;
+    private final long shutdownTimeoutMS;
+
+    public GRPCAgentClient(Format<com.expedia.open.tracing.Span> format, ManagedChannel channel, SpanAgentStub stub, long shutdownTimeoutMS) {
+        this.format = format;
+        this.channel = channel;
+        this.stub = stub;
+        this.shutdownTimeoutMS = shutdownTimeoutMS;
+    }
+
+    @Override
+    public boolean send(Span span) throws ClientException {
+        stub.dispatch(format.format(span), new StreamObserver<DispatchResult>() {
```

Comment:
OK as is, but I've often had trouble with high volume services that follow this anonymous inner class pattern that creates a new stateless object that gets thrown on the garbage heap after it finishes. And I attended a session at JavaOne a few years ago where the presenter, who worked for Oracle, asked who had faced this issue and solved it by de-anonymizing the class to not create a new instance every time.